examples/two_dimensional_turbulence.jl

```diff
@@ -0,0 +1,58 @@
+# # Two dimensional turbulence example
+#
+# In this example, we initialize a random velocity field and observe its viscous,
+# turbulent decay in a two-dimensional domain.
+
+using Oceananigans, PyPlot, Statistics
+
+# In addition to importing plotting and statistics packages, we import
+# some types and functions from `Oceananigans` that will aid in the calculation
+# and visualization of voriticty.
+
+using Oceananigans: Face, Cell
+using Oceananigans.TurbulenceClosures: ∂x_faa, ∂y_afa
+
+# We instantiate the model with a simple isotropic diffusivity
+
+model = Model(
+        grid = RegularCartesianGrid(N=(128, 128, 1), L=(2π, 2π, 2π)),
```

Comment:
Not sure if `Nz=1` works on the GPU but probably not a concern for this example.

Reply:
Yes, we need 3D examples to demonstrate GPU functionality right now. This will hopefully be the topic of a future PR.